### PR TITLE
Switch from S3 to Google Drive and automate with concourse

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/vk9FZSzS/204-work-out-if-how-we-can-automate-the-data-extraction-and-sharing-of-files

This reflects a change in the automation strategy where we have stopped using the Amazon S3 to upload files and are instead using Google Drive. It also covers a recent product development to provide a hashing functionality for email and phone fields.

This PR ends up doing a bunch of stuff (apologies it should have been separate PR's, I just kept adding to the same branch 😞 ). Things are explained in commits but here is a summary:

- Remove S3 integration and the s3_export task
- Creates a drive_export task and integrates with Google Drive API to share files
- Integrates with Notify to share links to files
- Includes two extra columns in Data Labs' export (hashed_email and hashed_phone) which use a SHA256 function and a salt (defined by SECRET_KEY env var) to anonymise the hash
- Adds a concourse pipeline file with the automatic job and the means to run a one off test job
- Outputs a slack message text file on the drive_export task which is used to create a dynamic Slack notification

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -67,9 +95,15 @@ GEM
...
     safe_yaml (1.0.5)
+    signet (0.14.0)
+      addressable (~> 2.3)
+      faraday (>= 0.17.3, < 2.0)
+      jwt (>= 1.5, < 3.0)
+      multi_json (~> 1.10)
     thread_safe (0.3.6)
     tzinfo (1.2.7)
       thread_safe (~> 0.1)
+    uber (0.1.0)
```
Comment (vanitabarrett) [inline Gemfile.lock:108]: 🚗

File: README.md [hunk 2]
```diff
@@ -18,6 +18,54 @@ bundle exec rake
...
+- `PERFORMANCE_ANALYST_DRIVE_FOLDER` - the file id of the Google Drive folder for
+  storing CSV exports for performance analysis
+- `PERFORMANCE_ANALYST_EMAIL_RECIPIENTS` - a comma separated list of email addresses
+  of performance analysts who will be emailed upon a successful export
+- `THIRD_PARTY_DRIVE_FOLDER` - the file id of the Google Drive folder for
+  storing CSV exports for the third party
+- `THIRD_PARTY_EMAIL_RECIPIENTS` - a comma separated list of email addresses
+  of third party colleagues who will be emailed upon a successful export
+- `SINCE_TIME` (optional) - defaults to "10:00", can be changed to alter the time
+  exports include data from. When this a relative time (for example "10:00") it
```
Comment (vanitabarrett) [inline README.md:65]:
```suggestion
  exports include data from. When this is a relative time (for example "10:00") it
```

File: README.md [hunk 3]
```diff
@@ -18,6 +18,54 @@ bundle exec rake
...
+  will be for the previous day, otherwise an absolute time can be set (for example
+  "2020-05-01 10:00") for a precise data export
+- `UNTIL_TIME` (optional) - defaults to "10:00", can be changed to alter the time
+  exports include data until. When this a relative time (for example "10:00") it
```
Comment (vanitabarrett) [inline README.md]:
```suggestion
  exports include data until. When this is a relative time (for example "10:00") it
```

**Comment by kevindew** [comment]
Thanks @issyl0 👍

**Pull request merged**